internal/gossiper/metrics.go

```diff
@@ -8,35 +8,65 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const namespace = "gossiper"
+
 type metrics struct {
-	txsReceived     prometheus.Counter
-	seenTxsReceived prometheus.Counter
-	txsGossiped     prometheus.Counter
+	txsReceived            prometheus.Counter
+	seenTxsReceived        prometheus.Counter
+	txsGossiped            prometheus.Counter
+	selectTxsToGossipCount prometheus.Counter
+	selectTxsToGossipSum   prometheus.Gauge
+	selectedTxsToGossip    prometheus.Counter
+	targetTxsSum           prometheus.Gauge
 }
 
 func newMetrics(r prometheus.Registerer) (*metrics, error) {
 	m := &metrics{
 		txsReceived: prometheus.NewCounter(prometheus.CounterOpts{
-			Namespace: "gossiper",
+			Namespace: namespace,
 			Name:      "txs_received",
 			Help:      "number of txs received over gossip",
 		}),
 		txsGossiped: prometheus.NewCounter(prometheus.CounterOpts{
-			Namespace: "gossiper",
+			Namespace: namespace,
 			Name:      "txs_gossiped",
 			Help:      "number of txs gossiped",
 		}),
 		seenTxsReceived: prometheus.NewCounter(prometheus.CounterOpts{
-			Namespace: "gossiper",
+			Namespace: namespace,
 			Name:      "seen_txs_received",
 			Help:      "number of txs received over gossip that were already seen",
 		}),
+		selectTxsToGossipCount: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "select_txs_to_gossip_count",
+			Help:      "number of times gossiper iterates mempool to select txs for gossip",
+		}),
+		selectTxsToGossipSum: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      "select_txs_to_gossip_sum",
```

Comment:
it's not obvious this + `target_txs_sum` are latency metrics from the naming - can we put `time` or something in the name to make it more clear?